index.js

```diff
@@ -16,9 +16,19 @@ function promisifyOwnProperties(obj) {
 }
 
 module.exports = obj => {
+  if (!obj) {
+    return obj;
+  }
+
+  const typeOfObj = typeof obj;
+
+  if (typeOfObj !== 'function' && (typeOfObj !== 'object' || Array.isArray(obj))) {
```

Comment:
Ah - so i guess you do need explicit handling for arrays just like functions, so that you can clone the array (with `.slice()`) and then assign over the promisified properties.

(however, let's count the number of promisified properties, and if it's zero, just return the original array directly?)